src/systemctl/systemctl-edit.c

```diff
@@ -405,7 +405,7 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
                 if (r < 0)
                         return r;
 
-                if (r == 0) {
+                if (r == 0 || !path) {
                         assert(!path);
 
```

Comment:
umm, this conditionalization in combination with the assert is weird. the assert says path is definitely NULL here. and that means the r == 0 test is entirely redundant?

Reply:
Uhhh yes though I had left in the `assert` in the event that `r == 0` somehow didn't end up with NULL path (as in the `else` case below where `r != 0` ended up with NULL path). I guess neither of that matters if I only check for `!path` and ignore `r`